### PR TITLE
Add TLA+ to supported languages

### PR DESCRIPTION
**Pull request by ahelwer**
Using [this](https://github.com/tlaplus-community/tree-sitter-tlaplus/) TLA+ tree-sitter grammar, locked to tagged commit v0.1.2.
Adds filetype detection, highlighting, folds, and injections for comments.

### CODE REVIEW

**Comment by ahelwer** [comment]
MacOS build is failing because of this issue: https://github.com/tree-sitter/tree-sitter/issues/1246

Will need to come up with a workaround.

**Review by stsewd — APPROVED** [review]

**Comment by ahelwer** [comment]
@stsewd are the CI failures due to my changes? At first glance the failures seem to be due to other language grammars.

**Comment by theHamsta** [comment]
@ahelwer no, current master fails with the same reason. I will disable the `required` passing in the Windows pipeline until this is fixed. It should not block unrelated PRs

File: queries/tlaplus/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  (temporal_exists)
+  (set_in)
+  (def_eq)
+  (gets)
+  (maps_to)
+  (all_map_to)
+  (case_box)
+  (case_arrow)
+  (label_as)
+] @keyword.function
```
Comment (theHamsta) [inline queries/tlaplus/highlights.scm:74]: Until now we only have keyword.function for keywords the introduce a function definition. Are those keywords  connected to similar concept? They seem to be able to introduce new clauses

**Review by theHamsta — APPROVED** [review]

File: queries/tlaplus/highlights.scm (hunk 1 above)
Comment (ahelwer) [reply queries/tlaplus/highlights.scm:74]: Hmmm, a couple of them roughly meet that definition but not all; I will experiment then make some changes.

**Review by vigoux — APPROVED** [review]

**Comment by vigoux** [comment]
Is this ready to be merged ?

**Comment by ahelwer** [comment]
@vigoux it is!

**Pull request merged**